fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/MeetingHandler.kt

```diff
@@ -0,0 +1,148 @@
+    val builder = MeetingBuilder()
+    builder
+        .setId(createMeeting.id)
+        .setCreation(createMeeting.creation)
+        .setStart(createMeeting.start)
+        .setEnd(createMeeting.end)
+        .setName(createMeeting.name)
+        .setLocation(createMeeting.location.orElse(""))
+        .setLastModified(createMeeting.creation)
+        .setModificationId("")
+        .setModificationSignatures(ArrayList())
+    val laoId = laoView.id
+    val meeting = builder.build()
+    witnessingRepo.addWitnessMessage(laoView.id, createMeetingWitnessMessage(messageId, meeting))
+    if (witnessingRepo.areWitnessesEmpty(laoId)) {
+      addMeetingRoutine(meetingRepo, laoId, meeting)
+    } else {
+      witnessingRepo.addPendingEntity(PendingEntity(messageId, laoId, meeting))
+    }
+  }
+
+  @Throws(UnknownLaoException::class)
+  fun handleStateMeeting(context: HandlerContext, stateMeeting: StateMeeting) {
+    val channel = context.channel
+    val messageId = context.messageId
+    Timber.tag(TAG).d("handleStateMeeting: channel: %s, name: %s", channel, stateMeeting.name)
+    val laoView = laoRepo.getLaoViewByChannel(channel)
+
+    // TODO: modify with the right logic when implementing the state functionality
+    val builder = MeetingBuilder()
+    builder
+        .setId(stateMeeting.id)
+        .setCreation(stateMeeting.creation)
+        .setStart(stateMeeting.start)
+        .setEnd(stateMeeting.end)
+        .setName(stateMeeting.name)
+        .setLocation(stateMeeting.location.orElse(""))
+        .setLastModified(stateMeeting.creation)
+        .setModificationId(stateMeeting.modificationId)
+        .setModificationSignatures(stateMeeting.modificationSignatures)
+    val laoId = laoView.id
+    val meeting = builder.build()
+    witnessingRepo.addWitnessMessage(laoView.id, stateMeetingWitnessMessage(messageId, meeting))
+    if (witnessingRepo.areWitnessesEmpty(laoId)) {
+      addMeetingRoutine(meetingRepo, laoId, meeting)
+    } else {
+      witnessingRepo.addPendingEntity(PendingEntity(messageId, laoId, meeting))
+    }
+  }
+
+  companion object {
+    private val TAG = RollCallHandler::class.java.simpleName
+
+    @JvmStatic
+    fun addMeetingRoutine(meetingRepository: MeetingRepository, laoId: String, meeting: Meeting) {
+      meetingRepository.updateMeeting(laoId, meeting)
+    }
+
+    @JvmStatic
+    fun createMeetingWitnessMessage(messageId: MessageID?, meeting: Meeting): WitnessMessage {
+      val message = WitnessMessage(messageId)
+      message.title = "The Meeting ${meeting.name} was created at ${Date(meeting.creation * 1000)}"
+      message.description =
+          """
+                   Mnemonic identifier :
+                   ${generateMnemonicWordFromBase64(meeting.id, 2)}
+                   
+                   ${
+        if (meeting.location.isEmpty()) "" else """
+     Location :
+     ${meeting.location}
+     
+     
+     """.trimIndent()
```

Comment:
put this in its own variable instead of embedding multi-line strings in multi-line strings, this is really not readable 😅